src/libcore/fmt/num.rs

```diff
@@ -188,11 +188,19 @@ static DEC_DIGITS_LUT: &[u8; 200] = b"0001020304050607080910111213141516171819\
 macro_rules! impl_Display {
     ($($t:ident),* as $u:ident via $conv_fn:ident named $name:ident) => {
         fn $name(mut n: $u, is_nonnegative: bool, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            // 2^128 is about 3*10^38, so 39 gives an extra byte of space
             let mut buf = [MaybeUninit::<u8>::uninit(); 39];
             let mut curr = buf.len() as isize;
             let buf_ptr = MaybeUninit::first_ptr_mut(&mut buf);
             let lut_ptr = DEC_DIGITS_LUT.as_ptr();
 
+            // SAFETY: Since `d1` and `d2` are always less than or equal to `198`, we
```

Comment:
Where did 198 come from?

Reply:
For the first case `d1 = (rem / 100) << 1` and `d2 = (rem % 100) << 1`, and `rem = n % 10000` so they're at most `99 * 2 = 198`. Same with the later `d1`s.

Reply:
Makes sense. I somehow managed to miss that % 100 is 0-99 vs. 0-100 or so.